### PR TITLE
[EPIC] Link Sharing

### PR DESCRIPTION
**Pull request by Tbaut**

### CODE REVIEW

**Comment by Tbaut** [comment]
The conflicts should be resolved in https://github.com/ChainSafe/ui-monorepo/pull/1653

File: packages/files-ui/src/Components/Modules/LinkSharingModule.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  }, [bucketDecryptionKey, encryptForPublicKey, publicKey])
+
+  useEffect(() => {
+    if(!jwt || !encryptedEncryptionKey || !!newBucket) return
+
+    filesApiClient.verifyNonce({ jwt, encryption_key: encryptedEncryptionKey })
+      .catch((e:any) => {
+        console.error(error)
+        setError(e.message)
```
Comment (FSM1) [inline packages/files-ui/src/Components/Modules/LinkSharingModule.tsx]:
```suggestion
        setError(e.error.message)
```

File: packages/files-ui/src/Components/Modules/LinkSharingModule.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      .catch(console.error)
+
+  }, [bucketDecryptionKey, encryptForPublicKey, publicKey])
+
+  useEffect(() => {
+    if(!jwt || !encryptedEncryptionKey || !!newBucket) return
+
+    filesApiClient.verifyNonce({ jwt, encryption_key: encryptedEncryptionKey })
+      .catch((e:any) => {
+        console.error(error)
```
Comment (FSM1) [inline packages/files-ui/src/Components/Modules/LinkSharingModule.tsx]:
```suggestion
        console.error(e)
```

**Review by FSM1 — CHANGES_REQUESTED** [review]
Left some minor suggestions, but besides that, it looks good to go 🥇

**Review by tanmoyAtb — APPROVED** [review]

**Comment by Tbaut** [comment]
Thanks @FSM1 I wonder if there's  better way to type these errors. I feel like we'll make the mistake with the `error.error.message` in the future.

**Review by FSM1 — APPROVED** [review]

**Comment by FSM1** [comment]
> Thanks @FSM1 I wonder if there's better way to type these errors. I feel like we'll make the mistake with the `error.error.message` in the future.

We really just need to get used to the API error response schema and always ensure to follow it. I've tried casting the object but it just adds unnecessary boilerplate.

**Comment by Tbaut** [comment]
Indeed, found nothing better than casting https://github.com/axios/axios/issues/3612#issuecomment-933263425

**Pull request merged**